drake/solvers/NloptSolver.cpp

```diff
@@ -0,0 +1,287 @@
+
+#include <stdexcept>
+#include <list>
+#include <vector>
+
+#include <nlopt.hpp>
+
+#include "NloptSolver.h"
+#include "Optimization.h"
+
+namespace Drake {
+namespace {
+Eigen::VectorXd MakeEigenVector(const std::vector<double>& x) {
+  Eigen::VectorXd xvec(x.size());
+  for (size_t i = 0; i < x.size(); i++) {
+    xvec[i] = x[i];
+  }
+  return xvec;
+}
+
+TaylorVecXd MakeInputTaylorVec(const Eigen::VectorXd& xvec,
+                               const VariableList& variable_list) {
+  size_t var_count = 0;
+  for (const DecisionVariableView& v : variable_list) {
+    var_count += v.size();
+  }
+
+  auto tx = initializeAutoDiff(xvec);
+  TaylorVecXd this_x(var_count);
+  size_t index = 0;
+  for (const DecisionVariableView& v : variable_list) {
+      this_x.segment(index, v.size()) = tx.segment(v.index(), v.size());
+      index += v.size();
+    }
+  return this_x;
+}
+
+double EvaluateCosts(const std::vector<double>& x,
```

Comment:
Yes.  The signatures to EvaluateCosts, EvaluateScalarConstraint, and EvaluateVectorConstraint are mandated by NLopt